client/components/ProductDetail/ProductDetail.jsx

```diff
@@ -13,159 +13,80 @@ import ShareButtons from './ShareButtons.jsx';
-    } = this.state;
-    return (
-      <Container id='ProductDetail'>
-        <h3>Product Details</h3>
-        {currentProduct && (
-          <ProductContainer>
-            <DisplayContainer>
-              <ImageGalleryContainer>
-                <ImageGallery currentStyle={currentStyle} changePhoto={this.changePhoto} />
-              </ImageGalleryContainer>
-              <ImageContainer>
-                <ProductImage currentPhoto={currentPhoto} />
-              </ImageContainer>
-            </DisplayContainer>
-            <ProductInfoContainer>
-              <ReviewsContainer>
-                <Stars number={rating} />
-                <ScrollToReviews allRatings={allRatings} />
-              </ReviewsContainer>
-              <br></br>
-              <div>
-                <b>Category:</b> {currentProduct.category}
-              </div>
-              <p>
-                <b>Product Name: </b>
-                {currentProduct.name}
-              </p>
+  }, [product.id]);
+
+  return (
+    <Container id='ProductDetail'>
+      <h3>Product Details</h3>
+      <ProductContainer>
+        {style && (
+          <DisplayContainer>
+            <ImageGalleryContainer>
+              <ImageGallery currentStyle={style} changePhoto={(photo) => setPhoto(photo)} />
+            </ImageGalleryContainer>
+            <ImageContainer>{photo && <ProductImage currentPhoto={photo} />}</ImageContainer>
+          </DisplayContainer>
+        )}
+        <ProductInfoContainer>
+          <ReviewsContainer>
+            <Stars number={rating} />
+            <ScrollToReviews allRatings={allRatings} />
+          </ReviewsContainer>
+          <br></br>
+          <div>
+            <b>Category:</b> {product.category}
+          </div>
+          <p>
+            <b>Product Name: </b>
+            {product.name}
+          </p>
+          {style && (
+            <>
               <p>
                 <b>Style: </b>
-                {currentStyle.name}
+                {style.name}
               </p>
-              <RenderPrice currentStyle={currentStyle} />
-              <StyleSelector productStyles={productStyles} changeStyle={this.changeStyle} />
-              <Cart>
-                <UpdateCart
-                  currentStyle={currentStyle}
-                  changeSize={this.changeSize}
-                  selectedSize={selectedSize}
-                  changeQuantity={this.changeQuantity}
-                  addToCart={this.addToCart}
-                />
-              </Cart>
-              <ShareButtons/>
-              {/* {currentProduct && <ProductDescription currentProduct={currentProduct} />} */}
-            </ProductInfoContainer>
-          </ProductContainer>
-        )}
-        <br></br>
-        {currentProduct && <ProductDescription currentProduct={currentProduct} />}
-        <br></br>
-      </Container>
-    );
-  }
-}
+
+              {/* TODO - Refactor this to accept default price if there styles.length === 0 */}
```

Comment:
great job TODOing. make sure this is transposed to a ticket in your team's system and then come back with a ticket ID or something to map the two